.github/workflows/analysis_workflow.yml

```diff
@@ -0,0 +1,293 @@
+      CMAKE_CXX_COMPILER_LAUNCHER: sccache
+      ARCTIC_CMAKE_PRESET: linux-debug
+      CIBW_ENVIRONMENT_PASS_LINUX: SCCACHE_GHA_VERSION ACTIONS_CACHE_URL ACTIONS_RUNTIME_TOKEN VCPKG_INSTALLATION_ROOT
+        VCPKG_BINARY_SOURCES VCPKG_NUGET_USER VCPKG_NUGET_TOKEN VCPKG_MAN_NUGET_USER VCPKG_MAN_NUGET_TOKEN
+        CMAKE_C_COMPILER_LAUNCHER CMAKE_CXX_COMPILER_LAUNCHER CMAKE_BUILD_PARALLEL_LEVEL ARCTIC_CMAKE_PRESET
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Get number of CPU cores
+        uses: SimenB/github-actions-cpu-cores@v1
+        id: cpu-cores
+
+      - name: Configure sccache
+        uses: mozilla-actions/sccache-action@v0.0.3
+        with:
+          version: "v0.4.0"
+
+      - name: Extra envs
+        run: |
+          . build_tooling/vcpkg_caching.sh # Linux follower needs another call in CIBW
+          echo -e "VCPKG_BINARY_SOURCES=$VCPKG_BINARY_SOURCES
+          VCPKG_ROOT=$PLATFORM_VCPKG_ROOT" | tee -a $GITHUB_ENV
+          cmake -P cpp/CMake/CpuCount.cmake | sed 's/^-- //' | tee -a $GITHUB_ENV
+        env:
+          CMAKE_BUILD_PARALLEL_LEVEL: ${{vars.CMAKE_BUILD_PARALLEL_LEVEL}}
+
+      - name: Prepare C++ compilation env
+        run: . build_tooling/prep_cpp_build.sh 
+      
+      - name: CMake compile
+        uses: lukka/run-cmake@v10
+        with:
+          cmakeListsTxtPath: ${{github.workspace}}/cpp/CMakeLists.txt
+          configurePreset: ${{env.ARCTIC_CMAKE_PRESET}}
+          buildPreset: ${{env.ARCTIC_CMAKE_PRESET}}
+        env:
+          ARCTICDB_CODE_COVERAGE_BUILD: 1
+        
+      - name: Run C++ Tests
+        shell: bash -l {0}
+        run: |
+          cd cpp/out/linux-debug-build/
+          make -j ${{ steps.cpu-cores.outputs.count }} arcticdb_rapidcheck_tests
+          make -j ${{ steps.cpu-cores.outputs.count }} test_unit_arcticdb
+          ctest
+
+      - name: Set real storage variables
+        run: |
+          # Common
+          echo "ARCTICDB_PERSISTENT_STORAGE_TESTS=1" >> $GITHUB_ENV
+          echo "ARCTICDB_PERSISTENT_STORAGE_BRANCH_NAME=${{ github.run_id }}" >> $GITHUB_ENV
+          # S3 Specific
+          # TODO: Add these as GitHub variables
+          echo "ARCTICDB_REAL_S3_BUCKET=arcticdb-ci-test-bucket-01" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_ENDPOINT=s3.eu-west-2.amazonaws.com" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_REGION=eu-west-2" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_CLEAR=1" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_ACCESS_KEY=${{ secrets.AWS_S3_ACCESS_KEY }}" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_SECRET_KEY=${{ secrets.AWS_S3_SECRET_KEY }}" >> $GITHUB_ENV
+
+      - name: Test with pytest
+        shell: bash -l {0}
+        run: |
+          yum install nodejs npm -y
+          npm config set strict-ssl false
+          npm install -g azurite
+          /opt/python/cp36-cp36m/bin/python -m pip install --upgrade pip
```

Comment:
Shall we use Python 3.11? 3.6 is end of life and we will probably remove support for it once users have all moved away from it. So using 3.11 will be safer for longer.